src/AST/Class.cs

```diff
@@ -102,6 +102,9 @@ public class Class : DeclarationContext
         // True if the class represents a static class.
         public bool IsStatic;
 
+        // True if declared with 'struct' keyword
+        public bool DeclaredStruct;
```

Comment:
A good question. I think it it all works, you'd better combine them. Also, now that you're at it, please make it a property (best if automatic) and not a field. I know the rest are field but it's a bad practice and we should eliminate it over time.